frontend/projects/abtesting/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts

```diff
@@ -243,11 +243,24 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
   }
 
   validateConditionCodes(conditions: ExperimentCondition[]) {
+
+    let conditionUniqueErrorText = this.translate.instant('home.new-experiment.design.condition-unique-validation.text');
     const conditionCodes = conditions.map(condition => condition.conditionCode);
     if (conditionCodes.length !== new Set(conditionCodes).size) {
-      this.conditionCodeError = this.translate.instant('home.new-experiment.design.condition-unique-validation.text')
-    } else {
-      this.conditionCodeError = null;
+      this.conditionCodeError = conditionUniqueErrorText;
+    }  else {
+        this.conditionCodeError = null;
+    }
+  }
+
+  validatehasConditionCodeDefault(conditions: ExperimentCondition[]) {
+    let defaultKeyword = this.translate.instant('home.new-experiment.design.condition.invalid.text');
+    let defaultConditionCodeErrorText = this.translate.instant('home.new-experiment.design.condition-name-validation.text')
+    if (conditions.length >= 1 ) {
+      const conditionCodes = conditions.map(condition => condition.conditionCode);
+      conditionCodes.forEach(conditioncode => {
+        conditioncode && conditioncode.toUpperCase() === defaultKeyword ? this.conditionCodeError = defaultConditionCodeErrorText : this.conditionCodeError = null;
```

Comment:
This logic seems wrong. Can we reverify this?

Reply:
You are correct. Nice catch. I did take care of this in the backend check but forgot the same in frontend one. Now it is handled properly.
<img width="807" alt="Screenshot 2021-06-17 at 7 01 34 PM" src="https://user-images.githubusercontent.com/33730817/122406616-9c5c2900-cf9e-11eb-9d15-79a75a16435d.png">